packages/grid/src/useGrid.ts

```diff
@@ -5,55 +5,27 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import { useState, HTMLProps, HTMLAttributes, KeyboardEventHandler } from 'react';
-import { composeEventHandlers, KEYS } from '@zendeskgarden/container-utilities';
+import { useState, KeyboardEventHandler, useCallback, useMemo } from 'react';
+import { composeEventHandlers, KEYS, useId } from '@zendeskgarden/container-utilities';
+import { IUseGridProps, IUseGridReturnValue } from './types';
+import { getCellDown, getCellLeft, getCellRight, getCellUp, getId } from './utils';
 
 const GRID_KEYS = [KEYS.LEFT, KEYS.RIGHT, KEYS.UP, KEYS.DOWN, KEYS.HOME, KEYS.END];
 
-export interface IUseGridProps {
-  /** Determines if navigation uses right-to-left direction */
-  rtl?: boolean;
-  /** Enables wrapped keyboard navigation  */
-  wrap?: boolean;
-  /** Sets the two-dimension array to render the grid */
-  matrix: any[][];
-  /** Prefixes IDs for the grid and cells  */
-  idPrefix?: string;
-  /** Sets the focused row index in a controlled grid */
-  rowIndex?: number;
-  /** Sets the focused column index in a controlled grid */
-  colIndex?: number;
-  /** Sets the default focused row index in a uncontrolled grid */
-  defaultRowIndex?: number;
-  /** Sets the default focused column index in a uncontrolled grid */
-  defaultColIndex?: number;
-  /** Handles grid change event */
-  onChange?: (rowIndex: number, colIndex: number) => void;
-  /** The environment where the grid is rendered */
-  environment?: Document;
-}
-
-interface IGetGridCellProps extends HTMLProps<any> {
-  rowIdx: number;
-  colIdx: number;
-}
-
-export interface IUseGridReturnValue {
-  getGridCellProps: (options: IGetGridCellProps) => HTMLAttributes<any>;
-}
-
 export function useGrid({
   rtl,
   wrap,
   matrix,
   idPrefix,
-  onChange,
-  environment = document,
+  onChange = () => undefined,
```

Comment:
Do we need a default value here? Couldn't we use optional chaining instead on line 60 below?

Reply:
This is consistent with the prevailing pattern throughout `react-containers`.